xtable-api/src/main/java/org/apache/xtable/conversion/ExternalTable.java

```diff
@@ -44,7 +45,7 @@ class ExternalTable {
   protected final CatalogConfig catalogConfig;
 
   /** Optional, additional properties that can be used to define interactions with the table */
-  protected final Properties additionalProperties;
```

Comment:
Why are we removing final ? Let's keep it immutable otherwise will be difficult to debug as this object is passed around multiple classes and function calls. If you want to change something you need to clone the object and change few properties.